### PR TITLE
marqeta-client-user-update: update function implemented with test.

### PR DESCRIPTION
**Pull request by jbearden**
Pull request for the Marqeta Node Client `create()` function.

### CODE REVIEW

File: src/user.ts [hunk 1]
```diff
@@ -146,6 +146,36 @@ export class UserApi {
     }
     return { success: !resp?.payload?.errorCode, body: { ...resp.payload } }
   }
+
+  /**
```
Comment (drbobbeaty) [inline src/user.ts:150]: Your other function blocks have `/*` and this one is `/**`... I don't care which, but pick one - Fair?

File: src/user.ts [hunk 2]
```diff
@@ -146,6 +146,36 @@ export class UserApi {
...
+   * in Marqeta, and return the account information. If no attributes are
+   * provided, a new User account will still be created and its
+   * token Id returned.
+   */
+  async create(user: Partial<User>): Promise<{
+    success: boolean,
+    body?: User,
+    error?: MarqetaError,
+  }> {
+    const searchOptions = snakecaseKeys(user)
```
Comment (drbobbeaty) [inline src/user.ts:161]: `searchOptions`? This really what you want to call this? Do you really even need it?

File: src/user.ts [hunk 3]
```diff
@@ -146,6 +146,36 @@ export class UserApi {
...
+    const resp = await this.client.fire('POST',
+      'users',
+      {},
```
Comment (drbobbeaty) [inline src/user.ts:164]: No query params is just as easily passed in as: `undefined`.

File: src/user.ts [hunk 4]
```diff
@@ -146,6 +146,36 @@ export class UserApi {
...
+      { ...searchOptions })
```
Comment (drbobbeaty) [inline src/user.ts:165]: Yeah... why isn't this just:

```typescript
  user,
)
```
?

**Review by drbobbeaty — CHANGES_REQUESTED** [review]
Just a few things...

File: src/user.ts (hunk 4 above)
Comment (jbearden) [reply src/user.ts:165]: right.  It has to be  `snakecaseKeys(user)`

**Review by drbobbeaty — APPROVED** [review]
Looks good

**Pull request merged**